File: BoardService/src/main/java/hana/simple/boardservice/global/exception/constant/RestControllerAdvice.java
package hana.simple.boardservice.global.exception.constant;

import hana.simple.boardservice.global.exception.ApplicationException;
import hana.simple.boardservice.global.response.APIResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.ExceptionHandler;

@org.springframework.web.bind.annotation.RestControllerAdvice
@Slf4j
public class RestControllerAdvice {

    @ExceptionHandler(ApplicationException.class)
    private APIResponse<String> exceptionHandler(ApplicationException error) {

        log.info("=== Application Exception occurred !! error : {}", error);
        return APIResponse.error(error);

    }

}

File: UserService/src/main/kotlin/hana/simple/userservice/global/exception/RestControllerAdvice.kt
package hana.simple.userservice.global.exception

import hana.simple.userservice.global.response.APIResponse
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class RestControllerAdvice {

    private val log = LoggerFactory.getLogger(RestControllerAdvice::class.java)

    @ExceptionHandler(ApplicationException::class)
    fun applicationHandler(error: ApplicationException) : APIResponse<String> {
        log.info("=== Application Exception occurred !! error : $error ===")
        return APIResponse.error(error)
    }

}
